handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -31,6 +32,9 @@
 /**
  * Triggers an {@link IdleStateEvent} when a {@link Channel} has not performed
  * read, write, or both operation for a while.
+ * Notice, that {@link IdleStateEvent} will be triggered for write operations only in the case when
+ * {@link DefaultChannelPromise} is used (i.e. {@link ChannelHandlerContext#write(Object)} or
+ * {@link ChannelHandlerContext#writeAndFlush(Object)}). IllegalArgumentException will be thrown otherwise.
```

Comment:
@maseev actually I think this is true for other handlers as well and was fixed as part of 4.1. The javadocs of the voidPromise() already state that it is some "advanced" feature which can only be used if you exactly know what you are doing.